server/src/com/cloud/servlet/ConsoleProxyServlet.java

```diff
@@ -414,7 +416,7 @@ private String composeThumbnailUrl(String rootUrl, VirtualMachine vm, HostVO hos
         return sb.toString();
     }
 
-    private String composeConsoleAccessUrl(String rootUrl, VirtualMachine vm, HostVO hostVo) {
+    private String composeConsoleAccessUrl(String rootUrl, VirtualMachine vm, HostVO hostVo, Boolean webSocketRequest) {
```

Comment:
why is this `Boolean` and not a `boolean`, in which case should you do a NPE check?